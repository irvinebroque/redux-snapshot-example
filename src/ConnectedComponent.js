import React, { Component } from 'react';
import { connect } from 'react-redux'
import AnotherConnectedComponent from './AnotherConnectedComponent'

class ConnectedComponent extends Component {
  render() {
    return (
      <div>
        <div>some text</div>
        <AnotherConnectedComponent/>
      </div>
    );
  }
}

const mapStateToProps = () => {
  return {}
}

export default connect(mapStateToProps)(ConnectedComponent)
