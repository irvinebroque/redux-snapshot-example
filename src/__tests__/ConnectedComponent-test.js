/* @flow */
import React from 'react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import renderer from 'react/lib/ReactTestRenderer'
import ConnectedComponent from '../ConnectedComponent'

function counter(state, action) {
  if (typeof state === 'undefined') {
    return 0
  }
  switch (action.type) {
    case 'INCREMENT':
      return state + 1
    case 'DECREMENT':
      return state - 1
    default:
      return state
  }
}

const store = createStore(counter)

test('ConnectedComponent renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <ConnectedComponent/>
    </Provider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
