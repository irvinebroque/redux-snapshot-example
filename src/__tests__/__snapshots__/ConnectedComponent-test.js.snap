exports[`test ConnectedComponent renders correctly 1`] = `
<div>
  <div>
    some text
  </div>
  <div>
    <div>
      some text
    </div>
  </div>
</div>
`;
